runtime/src/registry/mod.rs

```diff
@@ -0,0 +1,264 @@
+    }
+
+    fn create_new_registry_id() -> Result<RegistryId, dispatch::DispatchError> {
+        let id = <RegistryNonce>::get();
+
+        // TODO: Make a U160 type for RegistryId with the uint crate.
+        // Passing through U256 is inefficient and H160 is unneeded.
+        let mut res = Vec::<u8>::with_capacity(32);
+        unsafe { res.set_len(32); }
+        // U256 > H160 so no need for a checked_add
+        U256::from_big_endian(id.as_bytes())
+             .saturating_add(U256::one())
+             .to_big_endian(&mut res);
+
+        // Interpreted in big endian
+        let nplus1 = H160::from_slice(&res[0..20]);
+
+        // Update the nonce
+        <RegistryNonce>::put( nplus1 );
+
+        Ok(id)
+    }
+}
+
+// Implement the verifier registry. This module verifies data fields that are custom defined
+// by a registry and provided in the MintInfo during a mint invocation.
+impl<T: Trait> VerifierRegistry for Module<T> {
+    type AccountId    = <T as frame_system::Trait>::AccountId;
+    type RegistryId   = RegistryId;
+    type RegistryInfo = RegistryInfo;
+    type AssetId      = AssetId;
+    type AssetInfo    = <T as nft::Trait>::AssetInfo;
+    type MintInfo     = MintInfo<<T as frame_system::Trait>::Hash, H256>;
+
+    // Registries with identical RegistryInfo may exist
+    fn create_registry(mut info: Self::RegistryInfo) -> Result<Self::RegistryId, dispatch::DispatchError> {
+        // Generate registry id as nonce
+        let id = Self::create_new_registry_id()?;
+
+        // Create a field of the registry that is the registry id encoded with a prefix
+        let pre_reg = [NFTS_PREFIX, id.as_bytes()].concat();
+        info.fields.push(pre_reg);
+
+        // Insert registry in storage
+        Registries::insert(id.clone(), info);
+
+        Ok(id)
+    }
+
+    fn mint(caller: &<T as frame_system::Trait>::AccountId,
+            owner_account: &<T as frame_system::Trait>::AccountId,
+            asset_id: &Self::AssetId,
+            asset_info: T::AssetInfo,
+            mint_info: MintInfo<<T as frame_system::Trait>::Hash, H256>,
+    ) -> Result<(), dispatch::DispatchError> {
+        let (registry_id, token_id) = AssetIdRef::from(asset_id).destruct();
+        let registry_info = Registries::get(registry_id);
+
+        // Check that registry exists
+        ensure!(
+            Registries::contains_key(registry_id),
+            Error::<T>::RegistryDoesNotExist
+        );
+
+        // --------------------------
+        // Type checking the document
+
+        // The registry field must be a proof with its value as the token id.
+        // If not, the document provided may not contain the data and would
+        // be invalid. The registry field is always in the last place.
+        let registry_prop = &mint_info.proofs[ mint_info.proofs.len()-1 ].property;
+        ensure!(
+            // First check that property is long enough
+            registry_prop.len()                      >= 28 &&
```

Comment:
Since you have stored the registry property as a required attribute in the registry, why dont compare as a whole?
`registry_prop == reg_info.fields[0]`

Reply:
Good point. Is there any reason to worry about the padding here like there is in the solidity version?

Is it always an exact match or should it be
```registry_prop[..28] == reg_info.fields[n]```